lib/queue.js

```diff
@@ -82,7 +82,7 @@ var Queue = function Queue(name, url, opts){
   }
 
   if(_.isString(url)){
-    opts = _.defaults(redisOptsFromUrl, opts);
+    opts = {redis: _.defaults(redisOptsFromUrl(url), opts)};
```

Comment:
yes, that is completely wrong, I guess it was too late for coding when i did that :)